helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ZoneAccessor.java

```diff
@@ -0,0 +1,101 @@
+package org.apache.helix.rest.server.resources.helix;
+
+import com.codahale.metrics.annotation.ResponseMetered;
+import com.codahale.metrics.annotation.Timed;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import org.apache.helix.AccessOption;
+import org.apache.helix.BaseDataAccessor;
+import org.apache.helix.ConfigAccessor;
+import org.apache.helix.HelixDataAccessor;
+import org.apache.helix.HelixException;
+import org.apache.helix.PropertyPathBuilder;
+import org.apache.helix.model.ClusterConfig;
+import org.apache.helix.rest.common.HttpConstants;
+import org.apache.helix.rest.server.filters.ClusterAuth;
+import org.apache.helix.rest.server.filters.NamespaceAuth;
+import org.apache.helix.rest.server.json.cluster.ClusterTopology;
+import org.apache.helix.rest.server.service.ClusterService;
+import org.apache.helix.rest.server.service.ClusterServiceImpl;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+import org.apache.helix.zookeeper.zkclient.DataUpdater;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@ClusterAuth
+@Path("/clusters/{clusterId}/zones")
+public class ZoneAccessor extends AbstractHelixResource {
```

Comment:
It is better not have a separate resource for this. We already have REST resource clusterAccessor and there is an API for topology to get all zones / instances information.

Reply:
What is your concern aside from duplication?
Actually the opposite way looks more clear to me. We can extract the zone and topology related logic out of clusterAccessor. Think about this way, if zone is an entity, it deserves a separate accessor. And that could also be a step toward decoupling clusterConfig znode.

Reply:
Nope. This violets our convention on REST resource arrangement. When people looked at our REST user manual, they may get confused.

Reply:
Unless we totally get those things out of cluster config. Our REST aligns with ZK structure what we have.